src/filter.toit

```diff
@@ -0,0 +1,550 @@
+    return FilterNot_ filter
+
+  /**
+  Creates a raw filter.
+
+  The $raw string must be properly encoded and escaped. Use $encode to encode values.
+  Similarly, some characters in column names must be escaped. Use $encode_column to
+    encode column names.
+
+  # Example:
+
+  In this example we use the `any` and `all` modifiers to match any or all of the
+    given patterns on a specific column (without repeating the column name). These
+    modifiers are not directly supported and require the $Filter.raw
+    constructor to be used.
+
+  ```
+  middle_patterns := [
+    "Mid*",
+    "*e",
+  ]
+
+  // Escape the patterns.
+  // In this example this is not necessary, as no "dangerous" characters are used,
+  // but it's good practice to always escape values that are not known to be safe.
+  escaped_patterns := middle_patterns.map: Filter.encode it
+
+  // Matches any row where the first name starts with 'O' or 'P', the middle name
+  // matches all of the patterns in 'middle_patterns', and the last name starts
+  // with 'O' and ends with 'n'.
+  filters := [
+    Filter.raw "first_name=like(any).{O*,P*}",
+    Filter.raw "middle_name=like(all).{$(escaped_patterns.join ",")}",
+    Filter.raw "last_name=like(all).{O*,*n}",
+  ]
+  ```
+  */
+  constructor.raw raw/string:
+    return FilterRaw_ raw
+
+  /**
+  Builds the string representation of the filter that can be used in a PostgREST
+    request.
+
+  The result is already properly encoded and escaped.
+  */
+  to_string --nested/bool --negated/bool -> string
+
+class FilterBinary_ implements Filter:
+  column/string
+  op/string
+  value/any
+
+  constructor .column .op .value:
+
+  to_string --nested/bool --negated/bool -> string:
+    not_string := negated ? "not." : ""
+    column_string := Filter.encode_column column
+    encoded_value := Filter.encode value --may_quote=nested
+    return "$column_string$(nested ? "." : "=")$not_string$op.$encoded_value"
+
+class FilterIn_ implements Filter:
+  column/string
+  values/List
+
+  constructor .column .values:
+
+  to_string --nested/bool --negated/bool -> string:
+    not_string := negated ? "not." : ""
+    column_string := Filter.encode_column column
+    encoded_values := values.map: Filter.encode it --may_quote
+    joined := encoded_values.join ","
+    return "$column_string$(nested ? "." : "=")$not_string$Filter.IN.($joined)"
+
+
+class FilterArray_ implements Filter:
+  column/string
+  op/string
+  values/List
+
+  constructor .column .op .values:
+
+  to_string --nested/bool --negated/bool -> string:
+    not_string := negated ? "not." : ""
+    column_string := Filter.encode_column column
+    encoded_values := values.map: Filter.encode it --may_quote
+    joined := encoded_values.join ","
+    return "$column_string$(nested ? "." : "=")$not_string$op.{$joined}"
+
+class FilterRange_ implements Filter:
+  column/string
+  op/string
+  from/any  // Can be a num, or a date.
+  to/any    // Can be a num, or a date.
```

Comment:
```suggestion
  from/any  // Can be a num, or a Date.
  to/any    // Can be a num, or a Date.
```

Reply:
Leaving as is, as dates in strings are also allowed.